Std/Data/BitVec/Lemmas.lean

```diff
@@ -35,10 +35,18 @@ theorem testBit_toNat (x : BitVec w) : x.toNat.testBit i = x.getLsb i := rfl
 @[simp] theorem getLsb_ge (x : BitVec w) (i : Nat) (ge : i ≥ w) : getLsb x i = false := by
   let ⟨x, x_lt⟩ := x
   simp
-  apply Nat.testBit_lt_two
+  apply Nat.testBit_lt_two_pow
   have p : 2^w ≤ 2^i := Nat.pow_le_pow_of_le_right (by omega) ge
   omega
 
+theorem getLsb_eq_decide_and_getLsb {x : BitVec w} {i : Nat} :
+    getLsb x i = (decide (i < w) && getLsb x i) := by
+  if h : i < w then
+    simp [h]
+  else
+    have h' := Nat.ge_of_not_lt h
+    simp [decide_False, h, h']
```

Comment:
This is an usual style.  I'd instead expect something like `getLsb x i -> i < w` or the contra-positive.  Could you comment on the choice?

Reply:
Yeah, it is weird. The trick is where I use it in
```
@[simp] theorem getLsb_not {x : BitVec v} : (~~~x).getLsb i = (decide (i < v) && ! x.getLsb i) := by
  simp only [not_def, getLsb_xor, getLsb_allOnes]
  rw [getLsb_eq_decide_and_getLsb]
  -- case bash!
  cases decide (i < v) <;> simp
```
The point is that this introduces all the facts you want into the boolean expression, which can then be handled by boolean reasoning (in this case implemented via `cases` and `simp` because we don't have a sat solver).

Of course I could do it the tradional way with an implication, but it is extra work mixing the propositional and boolean approaches.

Reply:
How about we make this private for now?  I think the right approach is to have some form of programable [forward chaining](https://www.cs.utexas.edu/users/moore/acl2/manuals/current/manual/index-seo.php/ACL2____FORWARD-CHAINING), but I don't know of current Lean tactics that do that.

Reply:
I think this would be less objectionable if there were separate theorems saying `getLsb x i -> i < w` and `(a -> b) -> b && a = a` and `getLsb_not` just composed them.

Reply:
I think I'll do all these:
* split it into two steps
* make the lemma private
* leave a note at the point of use that a tactic is missing.

Reply:
I found a much better proof anyway.